### PR TITLE
156567661 more calendar validation

### PR DESCRIPTION
**Pull request by rckopteri**
# Added
* Better validations to trip calendars.

### CODE REVIEW

File: ote/src/cljs/ote/app/controller/route/route_wizard.cljs [hunk 1]
```diff
@@ -582,12 +585,12 @@
...
        transport-operator-id))
 
-(defn valid-calendar? [route-calendar]
-  (if (or (empty? route-calendar)
-          (and
-            (empty? (get route-calendar :rule-dates))
-            (empty? (get route-calendar ::transit/service-removed-dates))
-            (empty? (get route-calendar ::transit/service-rules)))) false true))
+(defn valid-calendar? [trip-calendar]
+  (if (or
```
Comment (tatut) [inline ote/src/cljs/ote/app/controller/route/route_wizard.cljs]: `(if cond false true)` is the same a `(not cond)`

File: ote/src/cljs/ote/app/controller/route/route_wizard.cljs [hunk 2]
```diff
@@ -512,14 +512,17 @@
...
-                            (fn [trips]
-                              ;; Update service-calendar indexes
-                              (mapv
-                                #(assoc % ::transit/service-calendar-idx
-                                          (nth cals-indices (::transit/service-calendar-idx %)))
-                                trips)))
-                    (dissoc :step :stops :new-start-time :edit-service-calendar))]
+                    (dissoc :step :stops :new-start-time :edit-service-calendar))
+          ;; Update calendar indexes if user has added calendars
+          route (if (and (not (nil? calendars)) (not (empty? calendars)))
```
Comment (tatut) [inline ote/src/cljs/ote/app/controller/route/route_wizard.cljs]: just say `(if (seq calendars) ...` to check if there are any calendars

File: ote/src/cljs/ote/app/controller/route/route_wizard.cljs [hunk 3]
```diff
@@ -615,3 +618,20 @@
 
 (defn valid-name [route]
   (if (empty? (get route ::transit/name)) false true))
+
+(defn valid-calendar-rule-dates? [data]
+  (let [check-rule (fn [rule]
+                     (and (not (empty? rule))
+                          (not (str/blank? (get rule ::transit/from-date)))
+                          (not (str/blank? (get rule ::transit/to-date)))))]
+    (if (empty? (::transit/service-rules data))
```
Comment (tatut) [inline ote/src/cljs/ote/app/controller/route/route_wizard.cljs]: `(and (seq (::transit/service-rules data))
    (every? ....)`

File: ote/src/cljs/ote/app/controller/route/route_wizard.cljs [hunk 4]
```diff
@@ -615,3 +618,20 @@
...
+      false
+      (every?
+        #(check-rule %)
+        (::transit/service-rules data)))))
+
+(defn empty-calendar-from-to-dates? [data]
+  (or
+    (str/blank? (get data ::transit/from-date))
```
Comment (tatut) [inline ote/src/cljs/ote/app/controller/route/route_wizard.cljs]: destructure and formatting

File: ote/src/cljs/ote/app/controller/route/route_wizard.cljs (hunk 1 above)
Comment (rckopteri) [reply ote/src/cljs/ote/app/controller/route/route_wizard.cljs]: Yes, but I return false from if, if one of those is empty. If I change this to use only "(or", then I would need to add "(not" to every check

**Pull request merged**